docs/www/guide-wasm-elastic.md

```diff
@@ -0,0 +1,218 @@
+/* function for validating record has the correct header key and value  */
+const containHeader = (key, value, record) =>
+  record.headers.some( header =>
+    header.headerKey.equals(Buffer.from(key)) &&
+    header.value.equals(Buffer.from(value))
+  )
+
+/* Transform function */
+transform.processRecord((recordBatch) => {
+  console.log("Applying wasm function")
+  return Promise.all(
+    recordBatch.records.map((record) => {
+      if (containHeader("logService", "elastic", record)) {
+        console.log("Index record value to Elastic Search")
+        return client.index({
+          index: "result_wasm",
+          type: "redpanda_wasm",
+          body: {
+            /*
+             index the record value into elastic search
+            */
+            record_message: record.value.toString()
+          }
+        })
+      }
+    })
+  ).then(() => {
+      /*
+       return a map with recordBatch indexed to elastic Search
+      */
+      const result = new Map();
+      result.set("elasticIndex", recordBatch)
+      /* 
+       Wasm processRecord function has to return a Map always,
+       although that map is empty map
+      */
+      return result
+  })
+  /*
+    Note: if the previous promise failed, the wasm errorHandler is
+    going to handle that error, in this case, Deregister policy will
+    remove this wasm function from wasm engine.    
+  */
+});
+
+exports["default"] = transform;
+```
+
+after applying the new logic, we need to bundle everything into a single file.
+For it, we can use 
+
+```bash
+#this command generate `dist/elastic.js` file.
+$ npm run build
+```
+
+Before deploying this wasm function, we need to create the `produce` topic
+The statically defined input topic that this wasm operator has defined in its metadata
+
+```bash
+rpk topic create produce
+```
+
+## Deploy wasm function to Redpanda
+
+we use `rpk wasm deploy ~//wasm_example/wasm/dist/elastic.js` in this case
+
+```bash
+$ rpk wasm deploy dist/elastic.js
+
+Sent record to partition 0 at offset 1 with timestamp 2021-02-01 18:18:15.734185538 -0500 -05 m=+0.053943881.
+```
+
+To verify that the coprocessor has successfully been deployed, you can look through the redpanda logs for a log line that looks like [this]
+for check it.
+
+on `redpanda logs`
+
+```bash
+INFO [shard 0] coproc - service.cc:101 - Request recieved to register coprocessor with id: 14103244480447969041
+```
+
+The wasm engine itself also has a logfile which by default will exist in `/var/lib/redpanda/coprocessor/logs/wasm`. To see the status of your transform reported by the wasm engine look for log lines like [this]
+
+```bash
+[FileManager] info: Detected new file in submit dir: /var/lib/redpanda/coprocessor/submit/elastic.js
+
+[FileManager] info: Initiating RPC call to redpandas coprocessor service at endpoint - enable_coprocessors with data: [object Object]
+TODO: improve [object Object]
+```
+
+Run data through the coprocessor
```

Comment:
> Should this be a title?

Yes, it is a title

Reply:
> Also be consistent with terms. Up until now, you've referred to it as "function". People will think a "coprocessor" or "transform" is a diffetent thing.

Done I'll refactor all name with, wasm script and wasm engine, in order to avoid ambiguous names

Reply:
Check with Alex & the sales team first!